File: dungeons-26.1.2/src/main/java/fishmod/features/dungeon/f7/MaxorTickTimer.java
package fishmod.features.dungeon.f7;

import config.practical.hud.HUDComponent;
import fishmod.utils.Location;
import fishmod.utils.config.values.Floor7;
import fishmod.utils.dungeon.Phase;
import fishmod.utils.events.Events;
import fishmod.utils.rendering.RenderUtils;
import net.minecraft.client.gui.GuiGraphicsExtractor;

/** Maxor (P1) tick timer — counts server ticks while in P1. Ported from blade-addons. */
public class MaxorTickTimer {

    private static int tick = 0;

    public static void init() {
        Events.ON_SERVER_TICK.register(() -> {
            if (Location.inDungeon() && Phase.inP1()) tick++;
            return false;
        });
        Events.ON_LOCATION_CHANGE.register(newLocation -> { tick = 0; return false; });
    }

    public static boolean display() {
        return Floor7.enableMaxorTickTimer && Location.inDungeon() && Phase.inP1();
    }

    public static void render(HUDComponent component, GuiGraphicsExtractor context) {
        RenderUtils.drawTimer(component, context, tick, 0xffffffff);
    }
}

File: dungeons-26.1.2/src/main/kotlin/fishmod/features/dungeon/f7/MaxorTickTimer.kt
package fishmod.features.dungeon.f7

import config.practical.hud.HUDComponent
import fishmod.utils.Location
import fishmod.utils.config.values.Floor7
import fishmod.utils.dungeon.Phase
import fishmod.utils.rendering.RenderUtils
import net.minecraft.client.gui.GuiGraphicsExtractor

/** Maxor (P1) tick timer — counts server ticks while in P1. Ported from blade-addons. */
object MaxorTickTimer {

    private val timer = TickTimer()

    @JvmStatic
    fun init() {
        timer.init(shouldCount = { Location.inDungeon() && Phase.inP1() })
    }

    @JvmStatic
    fun display(): Boolean {
        return Floor7.enableTickTimers && Floor7.enableMaxorTickTimer && Location.inDungeon() && Phase.inP1()
    }

    @JvmStatic
    fun render(component: HUDComponent, context: GuiGraphicsExtractor) {
        RenderUtils.drawTimer(component, context, timer.tick, 0xffffffff.toInt())
    }
}
